server/routes/rag.js

```diff
@@ -54,47 +59,59 @@ router.delete('/decks/:deckId', async (req, res) => {
 });
 
 router.get('/search', async (req, res) => {
+  const searchQuery = req.query.q || '';
+  if (!searchQuery.trim()) {
+    return res.status(400).json({ error: 'Search query is required' });
+  }
+  
+  const maxResults = parseInt(req.query.k) || 10;
+  const deckId = typeof req.query.deckId === 'string' ? req.query.deckId : undefined;
+  const exclude = typeof req.query.exclude === 'string' && req.query.exclude.length > 0
```

Comment:
I think we could do away with the length check, as combo os split and filter on empty string will give you `[]` anyway